archive/c/c/sleep-sort.c

```diff
@@ -0,0 +1,97 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <pthread.h>
+#include <unistd.h>
+
+pthread_mutex_t print_mutex = PTHREAD_MUTEX_INITIALIZER;
+int *global_sorted;
+int global_index = 0;
+int global_total;
+
+typedef struct {
+    int number;
+} ThreadPayload;
+
+void* sortNumber(void* args) {
+    ThreadPayload* payload = (ThreadPayload*) args;
+    const int number = payload->number;
+
+    usleep(number * 1000); // Sleep for number milliseconds
```

Comment:
I think you are going to need to sleep for longer than `number` milliseconds. Most implementations sleep for `number` seconds. Yes, it's slow, but it makes sure that the threads run in the right order.